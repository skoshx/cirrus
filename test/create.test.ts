// TODO: setup testing app environment before
// testsâ€¦

import { createApp, ready } from '..';

describe('create', () => {
  it('create > invalid options', async () => {
    await ready();
    // @ts-ignore
    expect(createApp('testapp', null)).rejects.toMatchSnapshot();
    // @ts-ignore
    expect(createApp('testapp', { port: 8080 })).rejects.toMatchSnapshot();
    expect(
      // @ts-ignore
      createApp('testapp', { port: 8080, appName: 'test' }),
    ).rejects.toMatchSnapshot();
  });
});
