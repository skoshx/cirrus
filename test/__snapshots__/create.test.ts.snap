// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create create > invalid options 1`] = `
[ZodError: [
  {
    "code": "invalid_type",
    "expected": "object",
    "received": "null",
    "path": [],
    "message": "Expected object, received null"
  }
]]
`;

exports[`create create > invalid options 2`] = `
[ZodError: [
  {
    "code": "invalid_type",
    "expected": "string",
    "received": "undefined",
    "path": [
      "appName"
    ],
    "message": "Required"
  },
  {
    "code": "invalid_type",
    "expected": "string",
    "received": "undefined",
    "path": [
      "errorFile"
    ],
    "message": "Required"
  },
  {
    "code": "invalid_type",
    "expected": "string",
    "received": "undefined",
    "path": [
      "logFile"
    ],
    "message": "Required"
  }
]]
`;

exports[`create create > invalid options 3`] = `
[ZodError: [
  {
    "code": "invalid_type",
    "expected": "string",
    "received": "undefined",
    "path": [
      "errorFile"
    ],
    "message": "Required"
  },
  {
    "code": "invalid_type",
    "expected": "string",
    "received": "undefined",
    "path": [
      "logFile"
    ],
    "message": "Required"
  }
]]
`;
